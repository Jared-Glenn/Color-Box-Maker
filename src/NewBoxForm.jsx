import React, { useState } from "react";

function NewBoxForm(props) {
    const INITIAL_STATE = {
        color: '',
        width: '',
        height: ''
    }
    const [formData, setFormData] = useState(INITIAL_STATE);
    const handleChange = (e) => {
        const { name, value } = e.target;
        setFormData(formData => ({
            ...formData,
            [name]: value
        }))
    }
    const handleSubmit = (e) => {
        e.preventDefault();
        props.addBox({ ...formData });
        setFormData(INITIAL_STATE)
    }
    
    return (
        <form onSubmit={handleSubmit}>
            <label htmlFor="color">Color: </label>
            <input
                id="color"
                type="text"
                name="color"
                placeholder="Color Name"
                value={formData.color}
                onChange={handleChange}
            />
            <label htmlFor="width">Width: </label>
            <input
                id="width"
                type="text"
                name="width"
                placeholder="Width in Pixels"
                value={formData.width}
                onChange={handleChange}
            />
            <label htmlFor="height">Width: </label>
            <input
                id="height"
                type="text"
                name="height"
                placeholder="Height in Pixels"
                value={formData.height}
                onChange={handleChange}
            />
            <button>Create Box</button>
        </form>
    )
}

export default NewBoxForm;