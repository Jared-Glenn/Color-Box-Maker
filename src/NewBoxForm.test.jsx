import { render } from "@testing-library/react";
import NewBoxForm from "./NewBoxForm";

// Smoke Test
it('renders without crashing', () => {
    render(<NewBoxForm />);
});

// Snapshot Test
it("matches the snapshot", () => {
    const {asFragment} = render(<NewBoxForm />);
                                    expect(asFragment()).toMatchSnapshot();
});
