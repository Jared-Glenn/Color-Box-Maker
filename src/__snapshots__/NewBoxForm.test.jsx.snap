// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<DocumentFragment>
  <form>
    <label
      for="color"
    >
      Color: 
    </label>
    <input
      id="color"
      name="color"
      placeholder="Color Name"
      type="text"
      value=""
    />
    <label
      for="width"
    >
      Width: 
    </label>
    <input
      id="width"
      name="width"
      placeholder="Width in Pixels"
      type="text"
      value=""
    />
    <label
      for="height"
    >
      Height: 
    </label>
    <input
      id="height"
      name="height"
      placeholder="Height in Pixels"
      type="text"
      value=""
    />
    <button>
      Create Box
    </button>
  </form>
</DocumentFragment>
`;
